src/test-utilities/enzyme.tsx

```diff
@@ -82,98 +73,85 @@ export function trigger(wrapper: AnyWrapper, keypath: string, ...args: any[]) {
   return returnValue;
 }
 
-/**
- * This is needed for updating the enzyme wrapper and react instance when we deeply change the context.
- * root.update() should work, but it doesn’t currently (see https://github.com/airbnb/enzyme/issues/1329).
- */
-export function forceUpdate(root: AnyWrapper) {
-  getInstance(root).forceUpdate();
-  root.update();
+function updateRoot(wrapper: AnyWrapper) {
+  (wrapper as any).root().update();
 }
 
-export function layeredContent(root: ReactWrapper<any, any>) {
-  const node = getInstance(root);
-  if (!node) {
-    return new ReactWrapper<any, any>([]);
-  }
+type AppContext = {
+  polaris: AppProviderContextType;
+  frame: FrameContext;
+};
 
-  const environment = {
-    childContextTypes: node.constructor.childContextTypes,
-    context: root.prop('context'),
-  };
+interface AppContextOptions {
+  app: AppContext;
+}
 
-  const layeredChildren = root.findWhere((wrapper: ReactWrapper<any, any>) => {
-    if (wrapper.length === 0) {
-      return false;
-    }
-    const wrapperNode = getInstance(wrapper);
-    return Boolean(wrapperNode && wrapperNode.renderLayer != null);
-  }) as any;
+interface MountWithAppProviderOptions {
+  context?: {
+    frame?: DeepPartial<FrameContext>;
```

Comment:
`DeepPartial` is used so we can only merge one branch of an object. For example, updating appBridge from undefined to X but leaving the rest of polaris intact.